src/pudl/transform/eia861.py

```diff
@@ -428,6 +541,27 @@
 ###############################################################################
 # EIA Form 861 Transform Helper functions
 ###############################################################################
+def _pre_process(df: pd.DataFrame) -> pd.DataFrame:
+    """Pre-processing applied to all EIA-861 dataframes.
+
+    * Standardize common NA values found in EIA spreadsheets.
+    * Drop the ``early_release`` column, which only contains non-null values when the
+      data is an early release, and we extract this information from the filenames, as
+      it's uniform across the whole dataset.
+    * Convert report_year column to report_date.
+    """
+    return (
+        fix_eia_na(df)
+        .drop(columns=["early_release"], errors="ignore")
+        .pipe(convert_to_date)
+    )
+
+
+def _post_process(df: pd.DataFrame) -> pd.DataFrame:
+    """Post-processing applied to all EIA-861 dataframes."""
+    return convert_cols_dtypes(df, data_source="eia")
```

Comment:
Could also use `Resource.enforce_schema()` here, but would need to pass in the table name. Not 100% sure which one would be preferable.

Reply:
`convert_cols_dtypes()` is a very old function that does a lot of nasty things, so I was afraid to swap it out for the simpler `apply_pudl_dtypes()` Definitely a longer conversation to be had (outside of this issue) about how / when to impose data types and do other data integrity checks outside of the DB or whatever storage medium the data is getting written out to.